test/asm/test.sh

```diff
@@ -47,13 +47,25 @@ warning: quote"file.asm(1): [-Wuser]
 EOF
 fi
 
+# Check whether to use '.simple.err' files if they exist
+# (rgbasm with pre-3.0 Bison just reports "syntax error")
+../../rgbasm -Weverything -o $o syntax-error.asm > $output 2> $errput
+cmp syntax-error.err $errput > /dev/null 2> /dev/null
```

Comment:
This can be simplified to
```bash
if cmp syntax-error.err $errput >/dev/null 2>/dev/null; then
    simple_error=
else
    echo (...)
    simple_error=.simple
fi
```
and then appending that to the two relevant variables.

Reply:
It needs to be conditionally appended based on `&& [ -e ${i%.asm}.simple.err ]`, and then `[ "$simple_error" -eq ".simple" ]` is not a simplification.

Reply:
I see. I would still prefer something in this vein, so that the check can be hoisted out of the `for variant` loop.

Reply:
Done.